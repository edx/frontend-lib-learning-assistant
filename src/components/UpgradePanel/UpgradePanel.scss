@use '../../utils/variables';

.upgrade-panel {
  height: 100%;

  overflow-y: auto;
  background-color: variables.$dark-green;

  h2 {
    font-size: 1.375rem;
  }

  .xpert-value-prop-check {
    color: variables.$accent-yellow;
  }

  .xpert-value-prop {
    margin-bottom: 1rem;
    font-size: 0.875rem;
  }
}
