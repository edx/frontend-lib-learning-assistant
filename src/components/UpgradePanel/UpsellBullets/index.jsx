import PropTypes from 'prop-types';
import {
  VerifiedCertBullet,
  UnlockGradedBullet,
  FullAccessBullet,
  XpertAccessBullet,
} from './UpsellBullets';

const UpsellContent = ({ isFBE }) => (
  <div role="list" className="pl-4 text-white pt-0 my-4">
    <XpertAccessBullet />
    <VerifiedCertBullet />
    {isFBE && (
      <>
        <UnlockGradedBullet />
        <FullAccessBullet />
      </>
    )}
  </div>
);

UpsellContent.defaultProps = {
  isFBE: false,
};

UpsellContent.propTypes = {
  isFBE: PropTypes.bool,
};

export default UpsellContent;
