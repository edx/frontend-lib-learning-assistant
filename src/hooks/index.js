export { default as useCourseUpgrade } from './use-course-upgrade';
export { default as useTrackEvent } from './use-track-event';
