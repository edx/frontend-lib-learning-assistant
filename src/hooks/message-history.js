/* eslint-disable import/prefer-default-export */
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getLearningAssistantMessageHistory } from '../data/thunks';

export const useMessageHistory = (courseId) => {
  const dispatch = useDispatch();
  const { isEnabled } = useSelector(state => state.learningAssistant);

  useEffect(() => {
    if (!courseId || !isEnabled) { return; }

    dispatch(getLearningAssistantMessageHistory(courseId));
  }, [dispatch, isEnabled, courseId]);
};
