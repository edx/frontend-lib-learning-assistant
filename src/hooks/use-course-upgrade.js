import { useContext } from 'react';
import { useModel } from '@src/generic/model-store'; // eslint-disable-line import/no-unresolved
import { useSelector } from 'react-redux';
import { CourseInfoContext } from '../context';

const millisecondsInOneDay = 24 * 60 * 60 * 1000; // hours*minutes*seconds*milliseconds

/**
 * @typedef AuditTrial
 * @type {object}
 * @property {string} startDate Date String of when the Trial started.
 * @property {string} expirationDate Date String of when the Trial ends.
 */

/**
 * @typedef CourseUpgradeInfo
 * @type {object}
 * @property {boolean} upgradeable Should this user see a trial/upgrade option?.
 * @property {number} [auditTrialLengthDays] Audit Trial full length in days.
 * @property {number} [auditTrialDaysRemaining] Remaining day for the current trial (if any).
 * @property {boolean} [auditTrialExpired] True means that the audit has been taken and expired.
 * @property {AuditTrial} [auditTrial] The Audit trial information. Undefined if there's no trial for this user.
 * @property {string} [upgradeUrl] URL to redirect the user in case there's intention to upgrade.
 */

/**
 * This hook returns related data for the Course Upgrade.
 *
 * @returns {CourseUpgradeInfo}
 */
export default function useCourseUpgrade() {
  const { courseId, isUpgradeEligible } = useContext(CourseInfoContext);
  const {
    offer,
    accessExpiration,
    datesBannerInfo,
  } = useModel('coursewareMeta', courseId);
  const { verifiedMode } = useModel('courseHomeMeta', courseId);
  const {
    auditTrialLengthDays,
    auditTrial,
  } = useSelector(state => state.learningAssistant);

  const upgradeUrl = offer?.upgradeUrl || verifiedMode?.upgradeUrl;

  if (!isUpgradeEligible || !upgradeUrl) { return { upgradeable: false }; }

  let auditTrialExpired = false;
  let auditTrialDaysRemaining;

  if (auditTrial?.expirationDate) {
    const auditTrialExpirationDate = new Date(auditTrial.expirationDate);
    auditTrialDaysRemaining = Math.ceil((auditTrialExpirationDate - Date.now()) / millisecondsInOneDay);

    auditTrialExpired = auditTrialDaysRemaining < 0;
  }

  const isFBE = !!accessExpiration && !!datesBannerInfo?.contentTypeGatingEnabled;

  return {
    upgradeable: true,
    auditTrialLengthDays,
    auditTrialDaysRemaining,
    auditTrialExpired,
    auditTrial,
    upgradeUrl,
    isFBE,
  };
}
