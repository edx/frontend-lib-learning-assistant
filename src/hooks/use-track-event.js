import { useContext } from 'react';
import { useModel } from '@src/generic/model-store'; // eslint-disable-line import/no-unresolved
import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
import { sendTrackEvent } from '@edx/frontend-platform/analytics';
import { CourseInfoContext } from '../context';

/**
 * @typedef Tracker
 * @type {object}
 * @property {(eventName: string, details?: object)} track Calls sendTrackEvent with user and course context.
 */

/**
 * This hook returns a track method to track events.
 *
 * @returns {Tracker}
 */
export default function useTrackEvent() {
  const { courseId, moduleId } = useContext(CourseInfoContext);
  const { org } = useModel('courseHomeMeta', courseId);
  const { userId } = getAuthenticatedUser();

  const track = (event, details) => {
    sendTrackEvent(event, {
      course_id: courseId,
      org_key: org,
      user_id: userId,
      module_id: moduleId,
      ...details,
    });
  };

  return { track };
}
