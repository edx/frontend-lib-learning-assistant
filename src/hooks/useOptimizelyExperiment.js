import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useDecision } from '@optimizely/react-sdk';
import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
import { setExperiment } from '../data/slice';

// We need this import to make sure Optimizely is instantiated.
import optimizelyInstance from '../data/optimizely'; // eslint-disable-line no-unused-vars

const useOptimizelyExperiment = (flag) => {
  const dispatch = useDispatch();
  const { userId } = getAuthenticatedUser();

  const [decision] = useDecision(flag, { autoUpdate: true }, { id: userId.toString() });
  const { active, variationKey } = decision || {};

  useEffect(() => {
    dispatch(setExperiment({ flag, active, variationKey }));
  }, [dispatch, flag, active, variationKey]);
};

export default useOptimizelyExperiment;
