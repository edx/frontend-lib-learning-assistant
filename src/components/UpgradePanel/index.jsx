import React from 'react';

import { useCourseUpgrade } from '../../hooks';
import UpsellContent from './UpsellBullets';
import UpgradeButton from '../UpgradeButton';

import './UpgradePanel.scss';

const UpgradePanel = () => {
  const { isFBE } = useCourseUpgrade();

  return (
    <div className="upgrade-panel d-flex flex-column align-items-stretch px-4 py-3">
      <h2 className="text-light-100">
        Upgrade this course
      </h2>
      <UpsellContent isFBE={isFBE} />
      <UpgradeButton includeLockIcon trackingEventName="edx.ui.lms.learning_assistant.expired_upgrade_click" />
    </div>
  );
};

export default UpgradePanel;
