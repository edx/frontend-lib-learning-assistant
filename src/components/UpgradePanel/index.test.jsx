import React from 'react';
import { render } from '../../utils/utils.test';
import { useCourseUpgrade, useTrackEvent } from '../../hooks';

import UpgradePanel from '.';

jest.mock('../../hooks', () => ({
  useCourseUpgrade: jest.fn(),
  useTrackEvent: jest.fn(),
}));

describe('UpgradePanel', () => {
  beforeEach(() => {
    useCourseUpgrade.mockReturnValue({ upgradeUrl: 'www.test.com' });
    useTrackEvent.mockReturnValue({ track: jest.fn() });
  });

  it('displays correct bullet points if not FBE', () => {
    useCourseUpgrade.mockReturnValue({
      upgradeUrl: 'www.test.com',
      isFBE: false,
    });

    expect(render(<UpgradePanel />)).toMatchSnapshot();
  });

  it('displays correct bullet points if FBE', () => {
    useCourseUpgrade.mockReturnValue({
      upgradeUrl: 'www.test.com',
      isFBE: true,
    });

    expect(render(<UpgradePanel />)).toMatchSnapshot();
  });
});
