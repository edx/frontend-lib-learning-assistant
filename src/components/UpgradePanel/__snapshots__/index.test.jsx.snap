// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpgradePanel displays correct bullet points if FBE 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="upgrade-panel d-flex flex-column align-items-stretch px-4 py-3"
      >
        <h2
          class="text-light-100"
        >
          Upgrade this course
        </h2>
        <div
          class="pl-4 text-white pt-0 my-4"
          role="list"
        >
          <div
            class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
            data-testid="xpert-access-bullet"
            role="listitem"
          >
            <span
              class="pgn__icon xpert-value-prop-check"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              Gain full access to 
              <span
                class="font-weight-bold"
              >
                Xpert Learning Assistant
              </span>
            </span>
          </div>
          <div
            class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
            data-testid="verified-cert-bullet"
            role="listitem"
          >
            <span
              class="pgn__icon xpert-value-prop-check"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              Earn a 
              <a
                class="inline-link-underline font-weight-bold text-white"
                href="http://localhost:18000/verified-certificate"
                rel="noopener noreferrer"
                target="_blank"
              >
                verified certificate
              </a>
               of completion to showcase on your resumé
            </span>
          </div>
          <div
            class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
            data-testid="unlock-graded-bullet"
            role="listitem"
          >
            <span
              class="pgn__icon xpert-value-prop-check"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              Unlock your access to all course activities, including 
              <span
                class="font-weight-bold"
              >
                graded assignments
              </span>
            </span>
          </div>
          <div
            class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
            data-testid="full-access-bullet"
            role="listitem"
          >
            <span
              class="pgn__icon xpert-value-prop-check"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              <span
                class="font-weight-bold"
              >
                Full access
              </span>
               to course content and materials, even after the course ends
            </span>
          </div>
        </div>
        <a
          class="trial-upgrade mt-3 btn btn-brand btn-block"
          data-testid="upgrade-cta"
          href="www.test.com"
        >
          <span
            class="pgn__icon my-0 mx-2"
            data-testid="lock-icon"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 8h-3V6.21c0-2.61-1.91-4.94-4.51-5.19A5.008 5.008 0 0 0 7 6h2c0-1.13.6-2.24 1.64-2.7C12.85 2.31 15 3.9 15 6v2H4v14h16V8zm-2 12H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                fill="currentColor"
              />
            </svg>
          </span>
          Upgrade now
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="upgrade-panel d-flex flex-column align-items-stretch px-4 py-3"
    >
      <h2
        class="text-light-100"
      >
        Upgrade this course
      </h2>
      <div
        class="pl-4 text-white pt-0 my-4"
        role="list"
      >
        <div
          class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
          data-testid="xpert-access-bullet"
          role="listitem"
        >
          <span
            class="pgn__icon xpert-value-prop-check"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Gain full access to 
            <span
              class="font-weight-bold"
            >
              Xpert Learning Assistant
            </span>
          </span>
        </div>
        <div
          class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
          data-testid="verified-cert-bullet"
          role="listitem"
        >
          <span
            class="pgn__icon xpert-value-prop-check"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Earn a 
            <a
              class="inline-link-underline font-weight-bold text-white"
              href="http://localhost:18000/verified-certificate"
              rel="noopener noreferrer"
              target="_blank"
            >
              verified certificate
            </a>
             of completion to showcase on your resumé
          </span>
        </div>
        <div
          class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
          data-testid="unlock-graded-bullet"
          role="listitem"
        >
          <span
            class="pgn__icon xpert-value-prop-check"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Unlock your access to all course activities, including 
            <span
              class="font-weight-bold"
            >
              graded assignments
            </span>
          </span>
        </div>
        <div
          class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
          data-testid="full-access-bullet"
          role="listitem"
        >
          <span
            class="pgn__icon xpert-value-prop-check"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            <span
              class="font-weight-bold"
            >
              Full access
            </span>
             to course content and materials, even after the course ends
          </span>
        </div>
      </div>
      <a
        class="trial-upgrade mt-3 btn btn-brand btn-block"
        data-testid="upgrade-cta"
        href="www.test.com"
      >
        <span
          class="pgn__icon my-0 mx-2"
          data-testid="lock-icon"
        >
          <svg
            aria-hidden="true"
            fill="none"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 8h-3V6.21c0-2.61-1.91-4.94-4.51-5.19A5.008 5.008 0 0 0 7 6h2c0-1.13.6-2.24 1.64-2.7C12.85 2.31 15 3.9 15 6v2H4v14h16V8zm-2 12H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              fill="currentColor"
            />
          </svg>
        </span>
        Upgrade now
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(Symbol.observable): [Function],
  },
  "unmount": [Function],
}
`;

exports[`UpgradePanel displays correct bullet points if not FBE 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="upgrade-panel d-flex flex-column align-items-stretch px-4 py-3"
      >
        <h2
          class="text-light-100"
        >
          Upgrade this course
        </h2>
        <div
          class="pl-4 text-white pt-0 my-4"
          role="list"
        >
          <div
            class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
            data-testid="xpert-access-bullet"
            role="listitem"
          >
            <span
              class="pgn__icon xpert-value-prop-check"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              Gain full access to 
              <span
                class="font-weight-bold"
              >
                Xpert Learning Assistant
              </span>
            </span>
          </div>
          <div
            class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
            data-testid="verified-cert-bullet"
            role="listitem"
          >
            <span
              class="pgn__icon xpert-value-prop-check"
            >
              <svg
                aria-hidden="true"
                fill="none"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span>
              Earn a 
              <a
                class="inline-link-underline font-weight-bold text-white"
                href="http://localhost:18000/verified-certificate"
                rel="noopener noreferrer"
                target="_blank"
              >
                verified certificate
              </a>
               of completion to showcase on your resumé
            </span>
          </div>
        </div>
        <a
          class="trial-upgrade mt-3 btn btn-brand btn-block"
          data-testid="upgrade-cta"
          href="www.test.com"
        >
          <span
            class="pgn__icon my-0 mx-2"
            data-testid="lock-icon"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 8h-3V6.21c0-2.61-1.91-4.94-4.51-5.19A5.008 5.008 0 0 0 7 6h2c0-1.13.6-2.24 1.64-2.7C12.85 2.31 15 3.9 15 6v2H4v14h16V8zm-2 12H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                fill="currentColor"
              />
            </svg>
          </span>
          Upgrade now
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="upgrade-panel d-flex flex-column align-items-stretch px-4 py-3"
    >
      <h2
        class="text-light-100"
      >
        Upgrade this course
      </h2>
      <div
        class="pl-4 text-white pt-0 my-4"
        role="list"
      >
        <div
          class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
          data-testid="xpert-access-bullet"
          role="listitem"
        >
          <span
            class="pgn__icon xpert-value-prop-check"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Gain full access to 
            <span
              class="font-weight-bold"
            >
              Xpert Learning Assistant
            </span>
          </span>
        </div>
        <div
          class="pgn__hstack pgn__stack-gap--3 align-items-start xpert-value-prop"
          data-testid="verified-cert-bullet"
          role="listitem"
        >
          <span
            class="pgn__icon xpert-value-prop-check"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2 4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span>
            Earn a 
            <a
              class="inline-link-underline font-weight-bold text-white"
              href="http://localhost:18000/verified-certificate"
              rel="noopener noreferrer"
              target="_blank"
            >
              verified certificate
            </a>
             of completion to showcase on your resumé
          </span>
        </div>
      </div>
      <a
        class="trial-upgrade mt-3 btn btn-brand btn-block"
        data-testid="upgrade-cta"
        href="www.test.com"
      >
        <span
          class="pgn__icon my-0 mx-2"
          data-testid="lock-icon"
        >
          <svg
            aria-hidden="true"
            fill="none"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 8h-3V6.21c0-2.61-1.91-4.94-4.51-5.19A5.008 5.008 0 0 0 7 6h2c0-1.13.6-2.24 1.64-2.7C12.85 2.31 15 3.9 15 6v2H4v14h16V8zm-2 12H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              fill="currentColor"
            />
          </svg>
        </span>
        Upgrade now
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(Symbol.observable): [Function],
  },
  "unmount": [Function],
}
`;
